pkgs/development/python-modules/duckduckgo-search/default.nix

```
@@ -1,57 +1,72 @@
-{ lib
-, aiofiles
-, buildPythonPackage
-, click
-, fetchFromGitHub
-, h2
-, httpx
-, lxml
-, pythonOlder
-, requests
-, setuptools
-, socksio
+{
+  lib,
+  buildPythonPackage,
+  click,
+  fetchFromGitHub,
+  pythonOlder,
+  setuptools,
+  orjson,
+  curl-cffi,
+  rustPlatform,
 }:
+let
+  curl-cffi_0_7_0 = curl-cffi.overrideAttrs (oldAttrs: rec {
+    version = "0.7.0b4";
+    src = fetchFromGitHub {
+      owner = "yifeikong";
+      repo = "curl_cffi";
+      rev = "v${version}";
+      hash = "sha256-txrJNUzswAPeH4Iazn0iKJI0Rqk0HHRoDrtTfDHKMoo=";
+    };
+  });
 
+  orjson_3_10_3 = orjson.overrideAttrs (oldAttrs: rec {
+    version = "3.10.3";
+    src = fetchFromGitHub {
+      owner = "ijl";
+      repo = "orjson";
+      rev = "refs/tags/${version}";
+      hash = "sha256-bK6wA8P/IXEbiuJAx7psd0nUUKjR1jX4scFfJr1MBAk=";
+    };
```

maybe consider using `finalAttrs` in `overrideAttrs`
https://nixos.org/manual/nixpkgs/stable/#sec-pkg-overrideAttrs

```suggestion
  orjson_3_10_3 = orjson.overrideAttrs (finalAttrs: oldAttrs: {
    version = "3.10.3";
    src = fetchFromGitHub {
      owner = "ijl";
      repo = "orjson";
      rev = "refs/tags/${finalAttrs.version}";
      hash = "sha256-bK6wA8P/IXEbiuJAx7psd0nUUKjR1jX4scFfJr1MBAk=";
    };
```